### DISCUSSION TITLE
What triggers address must be not nil?

### DISCUSSION THREAD

**Discussion by RailsCod3rFuture** [mongodb/mongo-ruby-driver#2608] [Q&A]
Having an issue where my mongodb application is stopping rails from showing actions...It states that `address must not be nil` -- everything else, seems to be working. I wanted to know what leads to this problem?

**Comment by alexbevi** [comment]
Hi @RailsCod3rFuture,

To help us troubleshoot this can you share the full error message, including stack trace when the error is raised?

**Comment by baranyeni** [comment]
@alexbevi there is no full trace error. it started to happen after upgrading ruby and rails. Only reproduces on Prod or locally with prod settings. The Mongoid.yml is identical in betweeen environments. I think it may be somehow related to Redis or SMTP connection but it's just a feeling. Thanks.

**Comment by baranyeni** [comment]
Ok I found the problem.

Probably your config was something like;
`config.cache_store = :redis_cache_store, { url: "#{ENV['REDIS_URL']}/cache" }`

It's no more allowed, you need to pass bare url without path or any string attached to it.

Doing it as shown below fixed it;

`config.cache_store = :redis_cache_store, { url: ENV['REDIS_URL'] }`

 it's the changes due to the redis configuration.

**Reply by alexbevi** [reply]
@baranyeni thanks for clarifying the underlying issue was with the Redis configuration. I'll close this discussion out as it appears you've solved this mystery :)

**Reply by baranyeni** [reply]
@alexbevi  Thank you so much :)
